Signal/src/Jobs/MessageFetcherJob.swift

```diff
@@ -110,38 +110,51 @@ public class MessageFetcherJob: NSObject {
             }
         }()
 
-        let envelopes: [SSKEnvelope] = messageDicts.compactMap { buildEnvelope(messageDict: $0) }
+        let envelopes: [SSKProtoEnvelope] = messageDicts.compactMap { buildEnvelope(messageDict: $0) }
 
         return (
             envelopes: envelopes,
             more: moreMessages
         )
     }
 
-    private func buildEnvelope(messageDict: [String: Any]) -> SSKEnvelope? {
+    private func buildEnvelope(messageDict: [String: Any]) -> SSKProtoEnvelope? {
         do {
             let params = ParamParser(dictionary: messageDict)
 
+            let builder = SSKProtoEnvelope.SSKProtoEnvelopeBuilder()
```

Comment:
I added in support for the builder pattern.  This ought to be very useful when we build complex, nested messages.